KHP3-4780: (feat) add mch clinical views for anc, labour and delivery, and postn…

ojwanganto opened the pull request:
…atal care

## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).


## Summary

This PR adds clinical views for MCH mother services - anc, labor and delivery, and postnatal care. 


## Screenshots

![image](https://github.com/palladiumkenya/kenyaemr-esm-3.x/assets/2161877/6582d9bd-4d37-424e-8134-67c450a39889)


## Related Issue

*None.*



## Other

donaldkibet reviewed (changes requested):
Thanks @ojwanganto Great start 🚀. Let some suggestions.

ojwanganto commented:
![image](https://github.com/palladiumkenya/kenyaemr-esm-3.x/assets/2161877/a7061c01-cf64-47fe-ab7b-4d95c2243c1a)
Updated UI. Changed from tabbed to content switcher

ojwanganto commented on packages/esm-patient-clinical-view-app/src/esm-mch-app/constants.ts:
```diff
@@ -0,0 +1,85 @@
+export const mnchEncounterType_UUID = '3eb24129-9f7d-4372-a14e-c2cf577e5db1';
```
I will clean this up better in a separate PR

ojwanganto commented on packages/esm-patient-clinical-view-app/src/data-table/o-table.component.tsx:
```diff
@@ -0,0 +1,56 @@
+  tableRows: any;
+}
+
+export const OTable: React.FC<TableProps> = ({ tableHeaders, tableRows }) => {
```
This is not entirely obs table. It is a wrapper around Encounter and observations.

donaldkibet commented:
Greate! 🤝 Thanks @ojwanganto
